python/plugins/processing/algs/qgis/Heatmap.py

```diff
@@ -79,74 +71,116 @@ def tags(self):
+
+        keys = list(self.KERNELS.keys())
+        kernel_shape_param = QgsProcessingParameterEnum(self.KERNEL,
+                                                        self.tr('Kernel shape'),
+                                                        keys,
+                                                        allowMultiple=False,
+                                                        defaultValue=0)
         kernel_shape_param.setFlags(kernel_shape_param.flags() | QgsProcessingParameterDefinition.FlagAdvanced)
         self.addParameter(kernel_shape_param)
-        decay_ratio = ParameterNumber(self.DECAY,
-                                      self.tr('Decay ratio (Triangular kernels only)'),
-                                      -100.0, 100.0, 0.0)
+
+        decay_ratio = QgsProcessingParameterNumber(self.DECAY,
+                                                   self.tr('Decay ratio (Triangular kernels only)'),
+                                                   QgsProcessingParameterNumber.Double,
+                                                   0.0, True, -100.0, 100.0)
         decay_ratio.setFlags(decay_ratio.flags() | QgsProcessingParameterDefinition.FlagAdvanced)
         self.addParameter(decay_ratio)
-        output_scaling = ParameterSelection(self.OUTPUT_VALUE,
-                                            self.tr('Output value scaling'), self.OUTPUT_VALUES)
+
+        keys = list(self.OUTPUT_VALUES.keys())
+        output_scaling = QgsProcessingParameterEnum(self.OUTPUT_VALUE,
+                                                    self.tr('Output value scaling'),
+                                                    keys,
+                                                    allowMultiple=False,
+                                                    defaultValue=0)
         output_scaling.setFlags(output_scaling.flags() | QgsProcessingParameterDefinition.FlagAdvanced)
         self.addParameter(output_scaling)
-        self.addOutput(OutputRaster(self.OUTPUT_LAYER,
-                                    self.tr('Heatmap')))
 
-    def name(self):
-        return 'heatmapkerneldensityestimation'
-
-    def displayName(self):
-        return self.tr('Heatmap (Kernel Density Estimation)')
+        self.addParameter(QgsProcessingParameterRasterDestination(self.OUTPUT_LAYER, self.tr('Heatmap')))
+        self.addOutput(QgsProcessingOutputRasterLayer(self.OUTPUT_LAYER, self.tr('Heatmap')))
 
     def processAlgorithm(self, parameters, context, feedback):
-        layer = QgsProcessingUtils.mapLayerFromString(self.getParameterValue(self.INPUT_LAYER), context)
+        layer = self.parameterAsVectorLayer(parameters, self.INPUT_LAYER, context)
 
-        radius = self.getParameterValue(self.RADIUS)
-        kernel_shape = self.getParameterValue(self.KERNEL)
+        radius = self.parameterAsDouble(parameters, self.RADIUS, context)
+        kernel_shape = self.parameterAsEnums(parameters, self.KERNEL, context)
```

Comment:
Should be parameterAsEnum - parameterAsEnums will return a list and is for multiple value parameters only